internal/bg/availability_queue.go

```diff
@@ -0,0 +1,64 @@
+package bg
+
+import (
+	"context"
+	"time"
+
+	"github.com/RHEnVision/provisioning-backend/internal/bus"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+)
+
+// buffered channel for incoming requests
+var asRequestChan = make(chan *bus.GenericMessage, availabilityStatusBatchSize)
+
+// EnqueueAvailabilityStatusRequest prepares a status request check to be sent in the next
+// batch to the platform bus. Messages can be delayed up to several seconds until sent.
+// The function can block if the enqueueing channel is full.
+func EnqueueAvailabilityStatusRequest(msg *bus.GenericMessage) {
+	asRequestChan <- msg
+}
+
```

Comment:
This is the msg being sent to Kafka through the asRequestChan? 
I would change the name of the channel.

Reply:
After a short chat, I understand. Renaming to `kafkaAvailabilityRequest`. :-D